src/extension/feature-pregeneralized/src/main/java/org/geoserver/data/gen/info/GeneralizationInfosProviderImpl.java

```diff
@@ -35,10 +44,9 @@ protected URL deriveURLFromSourceObject(Object source) throws IOException {
 
             GeoServerResourceLoader loader =
                     GeoServerExtensions.bean(GeoServerResourceLoader.class);
-            File f =
-                    Resources.find(
-                            Resources.fromURL(Files.asResource(loader.getBaseDirectory()), path),
-                            true);
+            Resource resource = loader.get(Paths.convert(path));
+            File f = Resources.find(resource);
```

Comment:
This all looks like it will work, but was not quite how I would of written it .. you are:
1. looking up a file location (which probably makes a resource)
2. converting the file location to a url
3. Then later in parseXML(URL) you have to look up resource store again to turn the URL into a resource, calling file() method to unpack the file to the local data directory staging location.

I think I would do all the resource store stuff once so it is clear, and tried for parse(Inputstream) method

Reply:
I tried to avoid creating the file but in this case the URL is rejected by the geotools code because no protocol is associated to the file.